hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py

```diff
@@ -0,0 +1,171 @@
+
+import bottle
+import typing as t
+from dataclasses import dataclass, asdict
+from hmalib.aws_secrets import AWSSecrets
+from threatexchange.api import ThreatExchangeAPI
+from hmalib.common.logging import get_logger
+from .middleware import jsoninator, JSONifiable
+from hmalib.common.config import HMAConfig
+from hmalib.common import config as hmaconfig
+from hmalib.lambdas.fetcher import ThreatExchangeConfig
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class Dataset(JSONifiable):
+    privacy_group_id: int
+    privacy_group_name: str
+    fetcher_active: bool
+    write_back: bool
+    in_use: bool
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "Dataset":
+        return cls(
+            d["privacy_group_id"],
+            d["privacy_group_name"],
+            d["fetcher_active"],
+            d["write_back"],
+            d["in_use"],
+        )
+
+
+@dataclass
+class DatasetsResponse(JSONifiable):
+    datasets_response: t.List[Dataset]
+
+    def to_json(self) -> t.Dict:
+        return {
+            "datasets_response": [
+                dataset.to_json() for dataset in self.datasets_response
+            ]
+        }
+
+
+@dataclass
+class SyncDatasetResponse(JSONifiable):
+    response: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+def sync_privacy_groups():
+    """
+    Implementation of the "sync_privacy_groups" function of threatexchange config.
+
+    1. call ThreatExchange API get_threat_privacy_groups_member
+    and get_threat_privacy_groups_owner to get the list of privacy groups
+
+    2. If the threat_updates_enabled is true, save it using config framework if it doesn't
+    exist in dynamoDB.
+
+    3. If one privacy group exists in dynamoDB but not in the result of get_threat_privacy_groups_owner
+    or get_threat_privacy_groups_member, update in_user to false
+    """
+    api_key = AWSSecrets.te_api_key()
+    api = ThreatExchangeAPI(api_key)
+    privacy_group_member_list = api.get_threat_privacy_groups_member()
+    privacy_group_owner_list = api.get_threat_privacy_groups_owner()
+    unique_privacy_groups = set(privacy_group_member_list + privacy_group_owner_list)
+    priavcy_group_id_in_use = set()
+
+    for privacy_group in unique_privacy_groups:
+        if privacy_group.threat_updates_enabled:
+            # HMA can only read from privacy groups that have threat_updates enabled.
+            # # See here for more details:
+            # https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-updates/v9.0
+            logger.info("Adding collaboration name %s", privacy_group.name)
+            priavcy_group_id_in_use.add(privacy_group.id)
+            config = ThreatExchangeConfig(
+                privacy_group.id,
+                # TODO Currently default to True for testing purpose,
+                # need to switch it to False before v0 launch
+                fetcher_active=True,
```

Comment:
This is an important configuration value. The team has not had a chance to reflect on whether this should be active by default or not. 

Can we extract this to a constant in the interim, please?